### PR TITLE
release: promote staging for Marinara Engine v2.4.6

### LINKED ISSUE #6242: [Feature]: Publish Marinara Engine v2.4.6 from tested staging
## What problem does this solve?

The tested staging changes and v2.4.6 release preparation need to reach stable users, including matching official Agents packages and installation/update artifacts.

## Proposed solution

Promote release-ready Marinara Engine staging to main as SpicyMarinara after checks and review. Finish #6241 first, verify Credits and version consistency, then publish v2.4.6 from the final main commit. Verify the source ZIP, Windows EXE, signed Android Termux-bootstrap APK and stable APK download alias, and both-architecture full/Lite Docker tags. Coordinate with Pasta-Devs/Marinara-Agents#873.

## Alternatives considered

Keep staging unreleased; the maintainer has now explicitly requested publication.

## Additional context

Owner: @SpicyMarinara, with Codex executing the maintainer's release request. Release preparation: #6233. What's New and animation: #6241. Docker branch tags must continue to distinguish main and staging. Real-device installation is not inferred from hosted build checks.

## Template check

- [ ] I DID NOT read this template and provide the requested details.

### PR DESCRIPTION
**Pull request by SpicyMarinara**
## Linked issue

Closes #6242

## Why this change

Publish the tested staging work as Marinara Engine v2.4.6, with the matching stable Agents catalog and downloadable installers. SpicyMarinara explicitly authorized both staging-to-main promotions and release publication.

## What changed

- Promote the existing staging history; release readiness is in #6233 and the final What's New announcement with supplied screenshots/videos and corrected memory wheel is in #6241.
- Synchronize v2.4.6 across application, Home, workspace and installers; Android version code 47. Move release changes under CHANGELOG v2.4.6 and keep Credits current (66 contributors).
- After this promotion, tag the exact final main commit and let the existing workflows publish the ZIP, Windows EXE, signed Termux-bootstrap APK and alias, and full/Lite multi-architecture container images.
- Coordinate stable package delivery with Pasta-Devs/Marinara-Agents#873. Agents retain independent package versions.

## Validation

Merged to main as `cc783dd194bacd97379191b9d490939afbe6759e`, with the same source tree as final staging `e0313a33caef973e645616419d11894f4d8c434e`. Tag v2.4.6 points to that exact main commit. [Release v2.4.6 is published](https://github.com/Pasta-Devs/Marinara-Engine/releases/tag/v2.4.6).

- Final [release regression matrix](https://github.com/Pasta-Devs/Marinara-Engine/actions/runs/35003059682): all 17 jobs pass on the first attempt; all 12 browser shards, 967 passed / 245 skipped / 0 failed; 282/282 Node scripts pass. Three embedded Bubblewrap sandbox proofs skip because Bubblewrap is unavailable.
- Final [platform matrix](https://github.com/Pasta-Devs/Marinara-Engine/actions/runs/35001941787): all 10 jobs pass, covering native Linux/macOS Intel/macOS ARM/Windows builds and launcher/update/shutdown paths, full/Lite amd64/arm64 container runtimes, Windows installer compile and Android signing guard.
- Local `pnpm check`, `pnpm localization:check`, final `pnpm version:check` and `pnpm credits:check` pass. Version is 2.4.6, Android version code 47, Credits current for 66 contributors.
- All five tag publishing workflows pass. Downloaded ZIP, EXE, versioned APK and APK alias match GitHub asset hashes and sizes. The source ZIP identifies the exact release commit and all package versions. The APKs are identical, validly signed with the previous published release's certificate, and report com.marinara.engine / 2.4.6 / code 47. The latest APK link resolves successfully to this release.
- All full and Lite stable Docker aliases resolve to the expected version and amd64/arm64 images. Main and staging branch tags point to their respective published commits. No local Docker layer downloads were needed to verify registry metadata.
- Agents #873 is merged to main as `a16819a2994070b53e0150122bacf6ed78c58334`; its 38-package catalogs and sampled changed artifacts match the promoted tree, and main CodeQL/quality checks pass.
- Engine post-merge [CodeQL](https://github.com/Pasta-Devs/Marinara-Engine/actions/runs/35004228985) and main code-quality checks also complete successfully.
- Both repositories' APIs report zero open Dependabot, code-scanning and secret-scanning alerts. Separate previously deferred quality/AI web dashboards are not represented by that claim.
- The release page credits 17 verified contributor accounts, including preserved patch coauthors and adopted PR submitters. All 17 avatars are present in the public GitHub Contributors footer.

[Published release, installer, Docker and contributor proof](https://gist.github.com/SpicyMarinara/067a9b89ec70dc62ea216d25a4196c6a).

CodeRabbit could not review this combined promotion because 823 eligible files exceed its configured 150-file limit and 300-file usage-priced maximum. SpicyMarinara explicitly approved a one-time exception relying on completed individual staging PR reviews for this promotion and Agents #873. Required CI passed before normal protected main merge. The focused shutdown-fixture correction #6246 received a complete final-head CodeRabbit review with no actionable findings; [controlled before/after proof](https://gist.github.com/SpicyMarinara/cbade75da97f256cdbb409cd0a32f247) documents its scope. New quality review comments were individually assessed and resolved with rationale; the suggested runtime rewrite of compile-time negative test calls was declined.

- [ ] `pnpm check` passes locally
- [ ] Container (Docker / Podman) built and ran without issue
- [ ] Ran the app, clicked through the changes manually
- [ ] Checked edge cases (light + dark mode, mobile viewport, empty states, error paths)
- [ ] Above manual verification completed (describe below)
- [ ] Read and followed `CONTRIBUTING.md`

### Manual verification notes

Browser automation uses isolated test data and does not establish real iPhone, Windows EXE installation or Android permission/Termux handoff behavior. Manually verify these device flows; hosted installer/build/runtime results will be linked explicitly. No user data or running installation is modified by these checks.

## Docs and release impact

- [ ] No docs changes needed
- [ ] Updated docs (README / CONTRIBUTING / android/README / CHANGELOG) as needed
- [ ] Translated docs on the `docs-i18n` branch updated to match, or a `[docs-i18n]` follow-up issue opened (see CONTRIBUTING.md § Translated documentation)
- [ ] Version/release files updated (only if this PR includes a version bump)

This is the stable publication handoff. Tag v2.4.6 is created only after main promotion. Main/staging Docker branch tags remain separate; tagged full images also update latest, and tagged Lite images update lite.

## UI evidence (if applicable)

What's New and wheel before/after evidence: https://gist.github.com/SpicyMarinara/07111eed47e93e5b7537df9bd5307d92
Character theme correction evidence: https://gist.github.com/SpicyMarinara/99cd073c1e5da2a839742671616bb4cc

### CODE REVIEW

**Comment by SpicyMarinara** [comment]
@coderabbitai review

**Comment by SpicyMarinara** [comment]
SpicyMarinara explicitly approved a one-time exception for the Engine #6243 and Agents #873 staging-to-main promotion reviews: use the completed individual staging PR reviews because CodeRabbit refuses the combined diffs above its hard file limit. Required CI and repository merge protections remain enforced. This authorizes the prepared release promotions and publication; no review completion is being fabricated.

**Comment by SpicyMarinara** [comment]
@coderabbitai review

File: scripts/regressions/fixtures/posix-terminal-shutdown.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        if finishing_socket:
+            finishing_socket.close()
+        if busy_socket:
+            busy_socket.close()
+        if master is not None:
+            os.close(master)
+        if status is None:
+            try:
+                os.killpg(child, signal.SIGKILL)
+            except ProcessLookupError:
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply scripts/regressions/fixtures/posix-terminal-shutdown.py:128]: Reviewed in the release gate. This is best-effort cleanup of the fixture child process group. ProcessLookupError specifically means the group already exited, which is the desired cleanup result; other errors still propagate and waitpid still reaps the child. The empty handler is intentional and introduces no shutdown or data-loss defect. Retaining the validated fixture; an additional explanatory comment is optional documentation cleanup.

File: scripts/regressions/fixtures/posix-terminal-shutdown.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                pass
+            os.waitpid(child, 0)
+        if server_pid:
+            try:
+                os.kill(server_pid, signal.SIGKILL)
+            except ProcessLookupError:
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply scripts/regressions/fixtures/posix-terminal-shutdown.py:134]: Reviewed in the release gate. This final best-effort kill catches only ProcessLookupError, meaning the fixture server is already gone. All other errors propagate; the preceding assertions separately require that the real server exited, the pending data was persisted, and the writer lease was released. No functional change is required. Retaining the validated fixture; a comment-only clarification is optional.

File: e2e/ui-state-fixture.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { UI_PERSISTENCE } from "../packages/client/src/lib/ui-persistence.js";
+import { seedUIState } from "./ui-state-fixture.js";
+
+test("UI fixtures use the live persistence contract and preserve existing preferences", async ({ page }) => {
+  // This is a local persistence contract test, not a cross-device settings sync test.
+  await page.route("**/api/app-settings/ui", async (route) => {
+    await route.fulfill({ json: { value: null } });
+  });
+  // Compile-time guards: unknown or renamed preferences must fail rather than silently seed nothing.
+  if (false) {
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply e2e/ui-state-fixture.e2e.ts:11]: Declining this suggested rewrite. These are explicitly documented compile-time negative controls: @ts-expect-error must keep checking the invalid preference name and invalid value type, without running either call. void does not make a call type-only; seedUIState immediately registers page.addInitScript that writes localStorage. The proposed calls would therefore seed invalid preferences at runtime. The deliberately unreachable branch is required by this existing proof pattern and is safe to retain.

File: packages/client/src/components/characters/CharacterLibraryView.tsx [hunk 4]
```diff
@@ -102,8 +96,20 @@ const LIBRARY_COPY: Record<CardLibraryKind, LibraryCopy> = {
...
+      personality: char.personality,
+      scenario: char.scenario,
+      first_mes: char.firstMessage,
+      creator_notes: char.creatorNotes,
+      tags: char.tags,
+      creator: char.creator,
+      character_version: char.version,
+      extensions: { fav: char.favorite, avatarCrop: char.avatarCrop, nameColor: char.nameColor },
+    };
+    return { ...char, parsed: (parsed as unknown as ParsedCharacterRow["parsed"]) ?? {} };
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply packages/client/src/components/characters/CharacterLibraryView.tsx:112]: Reviewed: parsed is constructed as an object literal here, so the fallback is indeed redundant. Keeping it does not change returned data or create a runtime/security defect. Treating this as optional behavior-neutral cleanup and retaining the already validated release application tree.

File: packages/client/src/components/chat/ChatSettingsDrawer.tsx [hunk 5]
```diff
@@ -6077,7 +6394,7 @@ export function ChatSettingsDrawer({
               label={localizeUi("ui.chat.chatsettingsdrawer.autonomousMessaging")}
               icon={<Bot size="0.875rem" />}
               help={localizeUi("ui.chat.chatsettingsdrawer.charactersCanMessageYouUnpromptedBasedOnTheirPersonality")}
-              initialOpen={initialSection === "autonomous"}
+              forceOpen={open && initialSection === "autonomous"}
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply packages/client/src/components/chat/ChatSettingsDrawer.tsx:6397]: Reviewed: the earlier if (!open) return null makes open true in this render path. The extra operand is redundant, and both forms produce the same forceOpen value. No behavior defect is present; treating this as optional cleanup and retaining the already validated release application tree.

File: packages/client/src/components/chat/ChatSettingsDrawer.tsx [hunk 6]
```diff
@@ -9100,20 +9527,26 @@ export function ChatSettingsDrawer({
           {!isConversation && import.meta.env.VITE_MARINARA_LITE !== "true" && (
             <Section
               id={`${chatMode}-memory-recall`}
+              forceOpen={open && initialSection === "memory-recall"}
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply packages/client/src/components/chat/ChatSettingsDrawer.tsx:9530]: Reviewed: the earlier if (!open) return null makes open true in this render path. The extra operand is redundant, and both forms produce the same forceOpen value. No behavior defect is present; treating this as optional cleanup and retaining the already validated release application tree.

File: packages/client/src/components/panels/CharactersPanel.tsx [hunk 7]
```diff
@@ -95,8 +98,20 @@ function getCharacterTags(char: ParsedCharacterRow): string[] {
...
+      personality: char.personality,
+      scenario: char.scenario,
+      first_mes: char.firstMessage,
+      creator_notes: char.creatorNotes,
+      tags: char.tags,
+      creator: char.creator,
+      character_version: char.version,
+      extensions: { fav: char.favorite, avatarCrop: char.avatarCrop, nameColor: char.nameColor },
+    };
+    return { ...char, parsed: (parsed as unknown as ParsedCharacterRow["parsed"]) ?? {} };
```
[the remark this answers is not in the document]
Comment (SpicyMarinara) [reply packages/client/src/components/panels/CharactersPanel.tsx:114]: Reviewed: parsed is constructed as an object literal here, so the fallback is indeed redundant. Keeping it does not change returned data or create a runtime/security defect. Treating this as optional behavior-neutral cleanup and retaining the already validated release application tree.
